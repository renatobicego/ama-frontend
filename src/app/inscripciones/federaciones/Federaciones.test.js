import '@testing-library/jest-dom/extend-expect'
import { cleanup, render, screen } from '@testing-library/react'
import Federaciones from './page'
import { useRouter } from 'next/navigation'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

describe('<Federaciones />', () => {

    let federaciones 
    let useRouterMock

    beforeEach(() => {
        
        // Mock useRouter, the Next.js router won't be available by default
        useRouterMock = jest.spyOn(require('next/navigation'), 'useRouter')
        useRouterMock.mockImplementation(() => ({
        back: jest.fn(),
        }))

        federaciones = render(<Federaciones />)
    })

    afterEach(() => {
        cleanup()
    })

    test('renders content', () => {
        expect(federaciones.container).toMatchSnapshot()

        // Renders Federaci贸n club link
        const club = screen.getByText('Federaci贸n Club')
        expect(club).toBeInTheDocument()

        // Renders Federaci贸n atleta link
        const atleta = screen.getByText('Federaci贸n Atleta')
        expect(atleta).toBeInTheDocument()

        //Renders imgs
        const clubImg = screen.getByAltText('Federacion Club')
        expect(clubImg).toBeInTheDocument()

        const federacionImg = screen.getByAltText('Federacion atleta')
        expect(federacionImg).toBeInTheDocument()
    })
})