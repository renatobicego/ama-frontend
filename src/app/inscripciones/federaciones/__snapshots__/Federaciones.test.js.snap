// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Federaciones /> renders content 1`] = `
<div>
  <main
    class="pt-[15vh] lg:pt-44 2xl:pt-52 pb-20"
  >
    <section
      class="size-section xl:mt-6"
    >
      <button
        class="align-middle select-none font-sans font-bold text-center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-2 px-4 rounded-lg hover:bg-red-500/10 active:bg-red-500/30 w-32 pl-2 mb-4 flex justify-start items-center gap-3 text-title"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="w-5"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.5 19.5L3 12m0 0l7.5-7.5M3 12h18"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        Volver
      </button>
      <h2
        class="text-title title-section text-left"
      >
        Federaciones
      </h2>
      <div
        class="items-center w-full md:w-[90%] flex justify-center gap-6 md:justify-between flex-wrap mt-10"
      >
        <a
          class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
          href="/inscripciones/federaciones/club"
        >
          <img
            alt="Federacion Club"
            class="object-cover object-center relative -z-10"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=%2Fimgs%2Ffederacion1.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimgs%2Ffederacion1.jpg&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Ffederacion1.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
          <h3
            class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
          >
            Federación Club
          </h3>
        </a>
        <a
          class="bg-tag-bg/50 relative w-4/5 md:w-[45%]"
          href="/inscripciones/federaciones/atleta"
        >
          <img
            alt="Federacion atleta"
            class="object-cover object-center relative -z-10"
            data-nimg="1"
            decoding="async"
            height="400"
            loading="lazy"
            src="/_next/image?url=%2Fimgs%2Ffederacion2.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimgs%2Ffederacion2.jpg&w=640&q=75 1x, /_next/image?url=%2Fimgs%2Ffederacion2.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
          <h3
            class="text-center font-title font-semibold text-base sm:text-lg md:text-xl lg:text-2xl w-52 md:w-96 text-white z-10 center-absolute"
          >
            Federación Atleta
          </h3>
        </a>
      </div>
    </section>
  </main>
</div>
`;
